drivers/gpu/drm/bridge/cdns-mhdp.c

```diff
@@ -577,10 +577,149 @@ static u8 eq_training_pattern_supported(struct cdns_mhdp_host host,
 	return fls(host.pattern_supp & sink.pattern_supp);
 }
 
+#define SET_HW_STATE(mhdp, newstate)					\
+	({								\
+		dev_dbg(mhdp->dev, "%s:%d hw_state %d => %d\n",		\
+			__func__, __LINE__, mhdp->hw_state, newstate);	\
+		WARN_ON(!spin_is_locked(&mhdp->start_lock));		\
+		mhdp->hw_state = newstate;				\
+	})
```

Comment:
Maybe this was useful when developing and debugging, but is there a need for this in the final version? I think the code is more readable if it does `mhdp->hw_state = newstate` rather than `SET_HW_STATE(mhdp, newstate)`.